compiler/src/symboltable.rs

```diff
@@ -99,6 +101,10 @@ pub struct Symbol {
     pub is_assigned: bool,
     pub is_parameter: bool,
     pub is_free: bool,
+
+    // indicates if the symbol gets a value assigned by a named expression in a comprehension
+    // this is required to correct the scope in the analysis.
+    pub is_assign_namedexpr_in_comprehension: bool,
```

Comment:
I guess this should be removed when the scoping is properly fixed? I cannot find the field here: https://docs.python.org/3/library/symtable.html?highlight=symboltable#symtable.SymbolTable

Reply:
This is required for scope extension, see my other comments. There can more optimal solutions, but I guess this more reqork of the symtable. Some of the context information could be gathered directly but this would increase the runtime.